paddle/fluid/inference/api/analysis_predictor.cc

```diff
@@ -269,6 +269,13 @@ void AnalysisPredictor::MkldnnPostReset() {
 #ifdef PADDLE_WITH_MKLDNN
   // In cache clearing mode.
   if (config_.mkldnn_cache_capacity_ > 0) {
+    if (VLOG_IS_ON(2)) {
+      auto shape_blob_size = static_cast<platform::MKLDNNDeviceContext *>(
+                                 (&platform::DeviceContextPool::Instance())
+                                     ->Get(platform::CPUPlace()))
+                                 ->GetShapeBlobSize();
+      CHECK_LE(shape_blob_size, config_.mkldnn_cache_capacity_);
```

Comment:
To avoid speed impact on inference deployment, we only do `CHECK_LE` when VLOG>=2 in debug mode.